test cases/common/262 internal dependency includes in checks/meson.build

```diff
@@ -0,0 +1,7 @@
+project('test 262', 'c')
+
+cc = meson.get_compiler('c')
+
+internal_dep = declare_dependency(include_directories: 'include')
```

Comment:
You miss `.as_system()` to make a useful unit test here, no? Also, how do you make a unit test that actually fails if it's not passed as system?